beacon_node/execution_layer/src/lib.rs

```diff
@@ -367,8 +405,8 @@ impl<T: EthSpec> ExecutionLayer<T> {
     }
 
     /// Cache a full payload, keyed on the `tree_hash_root` of its `transactions` field.
```

Comment:
I think this comment is incorrect. Isn't it keyed on the `tree_hash_root` of the entire header and not just the transactions?